src/software/thunderscope/robot_diagnostics/chicker.py

```diff
@@ -73,6 +110,91 @@ def __init__(self, proto_unix_io):
         self.geneva_value = 3
         self.power_value = 1
 
+    def send_command_and_timeout(self, command):
+        """
+        If buttons are enabled, sends a Kick command and disables buttons
+
+        Attaches a callback to re-enable buttons after 3 seconds
+
+        :param command: Command to send. One of ChickerCommandMode.KICK or ChickerCommandMode.CHIP
+        """
+        # if button is enabled
+        if self.kick_chip_buttons_enable:
+            # send kick primitive
+            self.send_command(command)
+            self.disable_kick_chip_buttons()
+
+            # set and start timer to re-enable buttons after 3 seconds
+            QTimer.singleShot(
+                3 * MILLISECONDS_PER_SECOND, self.enable_kick_chip_buttons
+            )
+
+    def disable_kick_chip_buttons(self):
+        """
+        Disables the buttons
+        """
+        self.kick_chip_buttons_enable = False
+
+    def enable_kick_chip_buttons(self):
+        """
+        If buttons should be enabled, enables them
+        """
+        if self.no_auto_selected:
+            self.kick_chip_buttons_enable = True
+
+    def set_should_enable_buttons(self, enable):
+        """
+        Changes if buttons are clickable or not based on boolean parameter
+
+        :param enable: boolean to indicate whether buttons should be made clickable or not
+        """
+        self.no_auto_selected = enable
+
+        if enable:
+            self.enable_kick_chip_buttons()
+        else:
+            self.disable_kick_chip_buttons()
+
+    def send_command(self, command):
+        """Sends a [auto]kick or [auto]chip primitive
+
+        :param command: enum int value to indicate what primitive to send
+        :returns: None
+
+        """
+
+        # gets slider values
+        geneva_value = self.geneva_slider.value()
+
+        power_value = self.power_slider.value()
+
+        power_control = PowerControl()
+        power_control.geneva_slot = geneva_value
+
+        # sends kick, chip, autokick, or autchip primitive
+        if command == ChickerCommandMode.KICK:
+            power_control.chicker.kick_speed_m_per_s = power_value
+        elif command == ChickerCommandMode.CHIP:
+            power_control.chicker.chip_distance_meters = power_value
+        elif command == ChickerCommandMode.AUTOKICK:
+            power_control.chicker.auto_chip_or_kick.autokick_speed_m_per_s = power_value
+        elif command == ChickerCommandMode.AUTOCHIP:
+            power_control.chicker.auto_chip_or_kick.autochip_distance_meters = (
+                power_value
+            )
+
+        # sends proto
+        self.proto_unix_io.send_proto(PowerControl, power_control)
+
+        time.sleep(0.001)
```

Comment:
Curious why this was added?

Reply:
The proto wasn't being sent to the buffer (two things cannot be sent at the same time to the buffer I guess?). 

Earlier this proto wasn't actually in the buffer so it kept spamming kicks (again)

not too clear why tho

Reply:
It's probably because `send_proto` puts the proto in the buffer without the arg `block` set to `true`:
`put_nowait` doc:
> Put an item into the queue without blocking. Only enqueue the item if a free slot is immediately available. Otherwise raise the Full exception.

So sleep is also not a perfect solution. A better solution would probably be to update the `ProtoUnixIO.send_proto` to take in an arg for blocking and a timeout, and `ThreadSafeBuffer.put` to also take a timeout. Then we can send the proto with the block command + a timeout. 
Something like:
https://github.com/UBC-Thunderbots/Software/blob/2330f76f64c53b1a23c58fb60b72d47482ba3da5/src/software/thunderscope/thread_safe_buffer.py#L39-L46

Reply:
done